src/monaco.contribution.ts

```diff
@@ -46,21 +57,46 @@ export class LanguageServiceDefaultsImpl implements monaco.languages.typescript.
 		}
 
 		if (this._extraLibs[filePath]) {
-			throw new Error(`${filePath} already a extra lib`);
+			this._extraLibs[filePath].version++;
+			this._extraLibs[filePath].content = content;
+		} else {
+			this._extraLibs[filePath] = {
+				content: content,
+				version: 1
+			};
+		}
+		if (this._eagerExtraLibSync) {
+			this.syncExtraLibs();
 		}
-
-		this._extraLibs[filePath] = content;
-		this._onDidChange.fire(this);
 
 		return {
 			dispose: () => {
-				if (delete this._extraLibs[filePath]) {
-					this._onDidChange.fire(this);

```

Comment:
I'm guessing that in the new code there's a case here where you still want to fire this?

Reply:
Yes, you still may want to remove an extra lib. However, removing it only triggers a resync and not a full worker reload.